src/magentic/prompt_chain.py

```diff
@@ -41,9 +43,13 @@ async def awrapper(*args: P.args, **kwargs: P.kwargs) -> Any:
                 chat = await Chat.from_prompt(
                     async_prompt_function, *args, **kwargs
                 ).asubmit()
+                calls = 0
                 while isinstance(chat.messages[-1].content, FunctionCall):
+                    if max_calls is not None and calls >= max_calls:
+                        raise PermissionError("max_calls reached")
```

Comment:
Could you create a new exception class for this please. Can add it to the top of this file.

```python
class MaxFunctionCallsError(Exception):
    """Raised when prompt chain reaches the max number of function calls."""
```

Reply:
Also would be useful to have the function name in the error message.

```suggestion
                        msg = f"Function {func.__name__} reached {max_calls} function calls"
                        raise MaxFunctionCallsError(msg)
```